dll/ntdll/ldr/ldrutils.c

```diff
@@ -35,7 +35,7 @@ LdrpAllocateUnicodeString(IN OUT PUNICODE_STRING StringOut,
 {
     /* Sanity checks */
     ASSERT(StringOut);
-    ASSERT(Length <= UNICODE_STRING_MAX_BYTES);
+    ASSERT(Length <= UNICODE_STRING_MAX_BYTES - sizeof(UNICODE_NULL));
```

Comment:
Why minus `sizeof(UNICODE_NULL)`?
There is code to handle the case where it's equal...

Reply:
OK, I'll revert it. I find the whole logic generally questionable, though.